### PR TITLE
Updated dependencies, dropped PHP 7.2 support and enabled Unit Tests in CI

### PR DESCRIPTION
**Pull request by driehle**
This PR updates all dependencies to the latest version. Since nearly all Laminas packages have dropped support for PHP 7.2 in favour of supporting PHP 8.0, this PR drops support for PHP 7.2 in DoctrineModule.

Since currently only coding standards are being checked (using PHP 7.2) in Gitlab Actions, I have updated the workflow to run unit tests using PHP 7.3 and 7.4 and to check coding standards using PHP 7.4. Running tests with PHP 8.0 is not possible yet, due to the dev dependencies to `laminas-mvc-console` and `laminas-test`. 

In regard to #725, end-users (i.e. not this library) need to specify the following in their `composer.json` for running DoctrineModule with PHP 8.0:

```
    "require": {
         "doctrine/doctrine-module": "^4.1.2"
    },
    "replace": {
        "laminas/laminas-cache-storage-adapter-apc": "*",
        "laminas/laminas-cache-storage-adapter-dba": "*",
        "laminas/laminas-cache-storage-adapter-memcache": "*",
        "laminas/laminas-cache-storage-adapter-memcached": "*",
        "laminas/laminas-cache-storage-adapter-mongodb": "*",
        "laminas/laminas-cache-storage-adapter-wincache": "*",
        "laminas/laminas-cache-storage-adapter-xcache": "*",
        "laminas/laminas-cache-storage-adapter-zend-server": "*"
    }
```

I suggest that these changes should be released as 4.2.0, i.e. a new minor release.

### CODE REVIEW

**Comment by driehle** [comment]
@TomHAnderson Since you already added commits to my PR, can we get this merged and released as 4.2.0?

**Review by TomHAnderson — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # DoctrineModule for Laminas
 
 [![Master Branch Build Status](https://secure.travis-ci.org/doctrine/DoctrineModule.png?branch=master)](http://travis-ci.org/doctrine/DoctrineModule) 
-[![Code Coverage](https://codecov.io/github/doctrine/DoctrineModule/coverage.svg?branch=4.1.x)](https://codecov.io/gh/doctrine/DoctrineModule/branch/4.1.x)
+[![Code Coverage](https://codecov.io/github/doctrine/DoctrineModule/coverage.svg?branch=4.2.x)](https://codecov.io/gh/doctrine/DoctrineModule/branch/4.2.x)
```
Comment (SenseException) [inline README.md:4]: Why putting 4.2 code coverage in the readme of the 4.1 branch?

Comment (TomHAnderson) [reply README.md:4]: Base changed to 4.2.x

**Review by SenseException — APPROVED** [review]

**Comment by driehle** [comment]
@greg0ire Is there something missing for getting this merged?

**Comment by greg0ire** [comment]
I don't think so, thanks!

**Pull request merged**